nucliadb_node/src/grpc/grpc_writer.rs

```diff
@@ -414,7 +414,7 @@ impl NodeWriter for NodeWriterGRPCDriver {
         &self,
         _request: Request<EmptyQuery>,
     ) -> Result<Response<NodeMetadata>, Status> {
-        let metadata = crate::node_metadata::NodeMetadata::new().await;
+        let metadata = crate::node_metadata::NodeMetadata::new(Arc::clone(&self.settings)).await;
```

Comment:
I wonder if we should use the inner pattern here to avoid all of the `Arc::clone` through the codebase. What I mean by inner pattern:
Instead of having `Arc<Settings>` in the codebase we define a private type in the `settings` module called  `InnerSettings`. This is basically just the current `Settings`, but without the methods. Then we define the type `Settings` in as: 
```rust
pub struct Settings {
      inner: Arc<Settings>,
}
```
Then we can implement cheap clonning for `Settings` and all the getters we want.